src/Collection/ArrayCollectionHelper.php

```diff
@@ -0,0 +1,95 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Arrays\Collection;
+
+use InvalidArgumentException;
+use Yiisoft\Arrays\Collection\Modifier\ModifierInterface\AfterMergeModifierInterface;
+use Yiisoft\Arrays\Collection\Modifier\ModifierInterface\BeforeMergeModifierInterface;
+
+final class ArrayCollectionHelper
+{
+    /**
+     * @param array|ArrayCollection ...$args
+     * @return ArrayCollection
+     */
+    public static function merge(...$args): ArrayCollection
```

Comment:
Is there a reason to return ArrayCollection here? Are we further interested in modifiers?

Reply:
> I think it can be part of the `ArrayCollection` itself. At least it will allow to expose less internal methods/data.

I like idea. Can make `ArrayCollection::mergeWith(...$args)` and hide several methods in `ArrayCollection`.

> And the merge method can be in ArrayHelper replacing current one.

In ArrayHelper::merge we can allow use collection, but result will be array, not collection.

> Is there a reason to return ArrayCollection here? Are we further interested in modifiers?

If need result collection can make so: `(new ArrayCollection())->mergeWith($a, $b, $c)`.